static/js/chart/draw_scatter.ts

```diff
@@ -550,6 +558,151 @@ function addRegressionLine(
   );
 }
 
+/**
+ * Adds text labels for a list of points
+ */
+function addTextLabels(
+  labelsGroup: d3.Selection<SVGGElement, any, any, any>,
+  xScale: ScatterScale,
+  yScale: ScatterScale,
+  points: Point[]
+): void {
+  labelsGroup
+    .attr("class", "dot-label")
+    .selectAll("text")
+    .data(points)
+    .enter()
+    .append("text")
+    .attr("dy", "0.35em")
+    .attr("x", (point) => xScale(point.xVal) + 7)
+    .attr("y", (point) => yScale(point.yVal))
+    .text((point) => point.place.name);
+}
+
+/**
+ * Gets at least MIN_HIGHLIGHT_POINTS points to highlight in each quadrant for a
+ * list of quadrants
+ */
+function getQuadrantHighlightPoints(
```

Comment:
we could split this up more into smaller fns but this is good for now!

Reply:
ok, will add a TODO